server/app/services/rpc_server.rb

```diff
@@ -13,30 +22,81 @@ def initialize(code, message, backtrace = nil)
     end
   end
 
-  ##
+  attr_reader :handlers
+
+  def initialize
+    @handlers = {}
+  end
+
+  # @param [Faye::Websocket] ws_client
   # @param [String] grid_id
   # @param [Array] message msgpack-rpc request array
   # @return [Array]
-  def self.handle_request(grid_id, message)
+  def handle_request(ws_client, grid_id, message)
     msg_id = message[1]
-    grid = Grid.find(grid_id)
-    return [1, msg_id, {error: 'grid not found'}, nil] if grid.nil?
-
     handler = message[2].split('/')[1]
     method = message[2].split('/')[2]
-    if klass = HANDLERS[handler]
+    if actor = handling_actor(grid_id, handler)
       begin
-        result = klass.new(grid).send(method, *message[3])
-        return [1, msg_id, nil, result]
+        result = actor.send(method, *message[3])
```

Comment:
Thinking... these are run as multiple concurrent Celluloid actor `rpc_server.async.handle_notification` tasks, and so this sync `actor.send(...)` is an Celluloid task scheduling point: https://github.com/celluloid/celluloid/wiki/Pipelining%20and%20execution%20modes

 Does this risk any concurrency issues between the shared `@handlers` state before and after this line? I guess maybe not, since the only operation we do after this line is the `@handlers[...].delete(...)`, which probably won't explode..?